invenio_app_ils/records/api.py

```diff
@@ -53,7 +74,112 @@ def create(cls, data, id_=None, **kwargs):
         return super(IlsRecord, cls).create(data, id_=id_, **kwargs)
 
 
-class Document(IlsRecord):
+class IlsRecordWithRelations(IlsRecord):
+    """Ils records class with relations."""
+
+    @staticmethod
+    def edition_relation():
+        """Get edition relation type."""
+        return current_app.config["EDITION_RELATION"]
+
+    @staticmethod
+    def language_relation():
+        """Get language relation type."""
+        return current_app.config["LANGUAGE_RELATION"]
+
+    @property
+    def editions(self):
+        """Get related editions."""
+        return self._get_related(self.edition_relation())
+
+    @property
+    def languages(self):
+        """Get related languages."""
+        return self._get_related(self.language_relation())
+
+    def node(self, relation_type):
+        """Get PIDNode for this record."""
+        return PIDNode(pid=self.pid, relation_type=relation_type)
+
+    def add_related_edition(self, record):
+        """Add record as a related edition record."""
+        relation = self.edition_relation()
+        self.add_related(
+            record,
+            relation,
+            parent_node=self._parent_node(relation)
+        )
+
+    def remove_related_edition(self, record):
+        """Remove related edition record."""
+        relation = self.edition_relation()
+        self.remove_related(
+            record,
+            relation,
+            parent_node=self._parent_node(relation)
+        )
+
+    def add_related_language(self, record):
+        """Add record as a related language record."""
+        relation = self.language_relation()
+        self.add_related(
+            record,
+            relation,
+            parent_node=self._parent_node(relation)
+        )
+
+    def remove_related_language(self, record):
+        """Remove related language record."""
+        relation = self.language_relation()
+        self.remove_related(
+            record,
+            relation,
+            parent_node=self._parent_node(relation)
+        )
+
+    def add_related(self, record, relation_type, parent_node=None):
+        """Add related record with a relation."""
+        record_node = record.node(relation_type)
+        if record_node.is_parent or record_node.is_child:
```

Comment:
As I understand here if a record has already relations cannot be added as a relation to another one right?

Reply:
Note these to two conditions are querying also db every time...

Reply:
Correct! Not sure otherwise how we would handle adding a record as an edition that already has other editions. Would we add those editions to the list? Feels like it could be messy. What do you think?

Yes, unfortunately it does. I hadn't planned for this from the start so the number of db calls are more now. Fortunately this is quite easy to optimize by storing some metadata in the record.

Reply:
Is this something that can be optimized eventually?

Reply:
Yes, we could store this information in the dump so we wouldn't have to hit the db.